src/test/java/com/fundynamic/d2tm/game/event/DrawableViewPortMoverListenerTest.java

```diff
@@ -0,0 +1,209 @@
+import org.mockito.runners.MockitoJUnitRunner;
+import org.newdawn.slick.Graphics;
+import org.newdawn.slick.SlickException;
+
+import static com.fundynamic.d2tm.Game.SCREEN_HEIGHT;
+import static com.fundynamic.d2tm.Game.SCREEN_WIDTH;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DrawableViewPortMoverListenerTest {
+    public static final float MOVE_SPEED = 2.0F;
+
+    public static final int ANY_COORDINATE_NOT_NEAR_BORDER = 100;
+    public static final float INITIAL_VIEWPORT_X = 4F;
+    public static final float INITIAL_VIEWPORT_Y = 4F;
+
+    public static int HEIGHT_OF_MAP = 20;
+    public static int WIDTH_OF_MAP = 26;
+
+    @Mock
+    private Viewport viewport;
+
+    private DrawableViewPort drawableViewPort;
+    private DrawableViewPortMoverListener listener;
+
+    private int renderAndUpdatedCalled;
+
+    @Before
+    public void setUp() throws SlickException {
+        Mockito.when(viewport.getMap()).thenReturn(new Map(null, WIDTH_OF_MAP, HEIGHT_OF_MAP));
+
+        drawableViewPort = new DrawableViewPort(viewport, new Vector2D<>(INITIAL_VIEWPORT_X, INITIAL_VIEWPORT_Y), mock(Graphics.class), MOVE_SPEED);
+        listener = new DrawableViewPortMoverListener(drawableViewPort);
+
+
+        renderAndUpdatedCalled = 0;
+    }
+
+    @Test
+    public void mouseHittingLeftBorderOfScreenMovesViewportToLeft() throws SlickException {
+        // moved to 2 pixels close to the left of the screen
+        listener.mouseMoved(3, ANY_COORDINATE_NOT_NEAR_BORDER, 2, ANY_COORDINATE_NOT_NEAR_BORDER);
+
+        Vector2D<Float> viewportVector = updateAndRenderAndReturnNewViewportVector();
+
+        Assert.assertEquals(INITIAL_VIEWPORT_X - MOVE_SPEED, viewportVector.getX(), 0.0001F);
+        Assert.assertEquals(INITIAL_VIEWPORT_Y, viewportVector.getY(), 0.0001F);
+    }
+
+    @Test
+    public void mouseHittingRightBorderOfScreenMovesViewportToRight() throws SlickException {
+        // moved to 2 pixels close to the right of the screen
+        listener.mouseMoved(SCREEN_WIDTH - 3, ANY_COORDINATE_NOT_NEAR_BORDER, SCREEN_WIDTH - 2, ANY_COORDINATE_NOT_NEAR_BORDER);
+
+        Vector2D<Float> viewportVector = updateAndRenderAndReturnNewViewportVector();
+
+        Assert.assertEquals(INITIAL_VIEWPORT_X + MOVE_SPEED, viewportVector.getX(), 0.0001F);
+        Assert.assertEquals(INITIAL_VIEWPORT_Y, viewportVector.getY(), 0.0001F);
+    }
+
+    @Test
+    public void mouseHittingTopOfScreenMovesViewportUp() throws SlickException {
+        // moved to 2 pixels close to the top of the screen
+        listener.mouseMoved(ANY_COORDINATE_NOT_NEAR_BORDER, 3, ANY_COORDINATE_NOT_NEAR_BORDER, 2);
+
+        Vector2D<Float> viewportVector = updateAndRenderAndReturnNewViewportVector();
+
+        Assert.assertEquals(INITIAL_VIEWPORT_X, viewportVector.getX(), 0.0001F);
+        Assert.assertEquals(INITIAL_VIEWPORT_Y - MOVE_SPEED, viewportVector.getY(), 0.0001F);
```

Comment:
You can combine them in one assert if you compare 2 vectors directly.
I can live with two asserts in a single test, though.